x-pack/legacy/plugins/apm/public/components/app/Settings/SettingsList.tsx

```diff
@@ -294,19 +313,21 @@ export function SettingsList() {
         </EuiCallOut>
 
         <EuiSpacer size="m" />
-
-        {status === 'success' && !hasConfigurations ? (
-          emptyStatePrompt
-        ) : (
-          <ManagedTable
-            noItemsMessage={<LoadingStatePrompt />}
-            columns={COLUMNS}
-            items={data}
-            initialSortField="service.name"
-            initialSortDirection="asc"
-            initialPageSize={50}
-          />
-        )}
+        {status === 'failure' ? failurePrompt : null}
+        {status === 'success' ? (
+          hasConfigurations ? (
+            <ManagedTable
+              noItemsMessage={<LoadingStatePrompt />}
+              columns={COLUMNS}
+              items={data}
+              initialSortField="service.name"
+              initialSortDirection="asc"
+              initialPageSize={50}
+            />
+          ) : (
+            emptyStatePrompt
+          )
+        ) : null}
```

Comment:
The previous `status === 'success' && !hasConfigurations` logic was rending `emptyStatePrompt` when `status` was `loading` so the new nested logic makes sure that nothing is rendered at that time. This is especially jarring if there's a failure message since it used to flash the `emptyStatePrompt` immediately prior to the failure message.

I'm open to alternatives for the nested ternary. :)

Reply:
What about a new component:
```ts
function Foo(status) {
  if (status === 'failure') {
    return failurePrompt;
  }

  if (status === 'success') {
    if (!hasConfigurations) {
      return emptyStatePrompt;
    }

    return (
      <ManagedTable
        noItemsMessage={<LoadingStatePrompt />}
        columns={COLUMNS}
        items={data}
        initialSortField="service.name"
        initialSortDirection="asc"
        initialPageSize={50}
      />
    );
  }
}
```

Reply:
created a new component for this called `SettingsList` (the old SettingsList is now just Settings/index since it renders the settings page.